### PR TITLE
Use FindPkgConfig and petsc.pc instead of FindPETSc.cmake closes #1913

### LINKED ISSUE #1913: Use FindPkgConfig and petsc.pc instead of FindPETSc.cmake
it is recommended to use petsc.pc and FindPkgConfig rather than FindPETSC.cmake to deal with non standard PETSc installation like in fedora

### LINKED ISSUE #1914: Port Feel++  to Fedora 36/37
Porting Feel++ to fedora requires a few changes:

- [x] #1913 
- [x] #1933
- [x] #1934
- [x] #1935
- [x] #1936
- [ ] #1937
- [x] #1981

### PR DESCRIPTION
**Pull request by prudhomm**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [ ] Have you successfully run the Feel++ testsuite with your changes locally?
- [x] Have you written Doxygen comments in your contribution ?

-----
- closes #1913 
- closes #1914

### CODE REVIEW

**Comment by prudhomm** [comment]
to do still
- [ ] #1914

**Comment by prudhomm** [comment]
@vincentchabannes feel++ compiles on fedora:37 (I haven't tested yet on fedora:36), several compilation fixes have been made due to clang 14.
Also the documentation of the toolbox applications have been updated and are now generated automatically

**Comment by prudhomm** [comment]
@vincentchabannes the PR builds fine. The remaining builds are due to changes to build fedora via github actions. The important build is this one: https://buildkite.com/feelpp/feelpp/builds/4655.
Please review

File: feelpp/cmake/modules/feelpp.module.petsc.cmake [hunk 1]
```diff
@@ -82,7 +85,7 @@ if ( PETSC_HAVE_MUMPS )
   set(FEELPP_HAS_MUMPS 1)
   set(FEELPP_PETSC_ENABLED_OPTIONS "${FEELPP_PETSC_ENABLED_OPTIONS} MUMPS")
 else()
-  message( FATAL_ERROR "PETSc should be compiled with MUMPS")
+  message( STATUS "PETSc should be compiled with MUMPS")
```
Comment (vincentchabannes) [inline feelpp/cmake/modules/feelpp.module.petsc.cmake:88]: PETSc should be compiled with Mumps

Comment (prudhomm) [reply feelpp/cmake/modules/feelpp.module.petsc.cmake:88]: unfortunately fedora petsc version is not compiled with mumps support

Comment (prudhomm) [reply feelpp/cmake/modules/feelpp.module.petsc.cmake:88]: maybe  replace with a warning ?

Comment (vincentchabannes) [reply feelpp/cmake/modules/feelpp.module.petsc.cmake:88]: I think no, an error will be preferred else we didn't ensure that all work correctly, this is the reason we have added this.
You can put a cmake variable (or fedora detection in cmake) if you want to use a petsc without mumps

Comment (prudhomm) [reply feelpp/cmake/modules/feelpp.module.petsc.cmake]: see #1981

**Comment by prudhomm** [comment]
the current build is here : https://buildkite.com/feelpp/feelpp/builds/4668

**Review by vincentchabannes — APPROVED** [review]

**Pull request merged**